### ISSUE TITLE
Making 'files' in 'tsconfig.json' optional

### ISSUE THREAD

**Issue by samvv** [TypeStrong/ts-loader#556]
Hi,

I'm implementing a system that packs a source directory that does not have a `tsconfig.json`, and therefore I need to point to a specific config file residing with the packer itself. Currently, it seems that this is not supported due to the `path.join` in [`findConfigFile`](https://github.com/TypeStrong/ts-loader/blob/f4d8e711b2193cec358f94adf92202f39b8b2473/src/config.ts#L75). Could this behaviour be changed so that it can accept absolute paths? If so, would you accept a pull request?

Regards,
Sam

**Comment by samvv** [comment]
Seems I was being too fast. The file resolves as it is supposed to; it is just not being correctly reported.

**Comment by johnnyreilly** [comment]
@jbrantly what's your view on this?  I seem to remember there being discussion around this in the past but I can't remember the conclusion...

**Comment by johnnyreilly** [comment]
Related to https://github.com/TypeStrong/ts-loader/issues/539 (as well as others no doubt)

**Comment by samvv** [comment]
Actually, now I come to think of it while working on it, the **real** issue I'm having is that `files` needs to be ignored, which is currently not possible. It throws the following error if I try to set it to an empty object, which is quite reasonable:

```
ERROR in /Users/samvv/Projects/engage2/proto2/packages/engagejs-platform/tsconfig.plugin.json
error TS18002: The 'files' list in config file 'tsconfig.json' is empty.
```

However, I know from the [wiki](https://github.com/Microsoft/TypeScript/wiki/Using-the-Compiler-API#incremental-build-support-using-the-language-services) that it should be possible, but I'm not sure how it would need to be translated to this project.

I might be wrong, but when I think about it, this seems more logical than providing yet _another_ entry point in `files`, while the real the webpack config containing the real entry point.

To sum it up: does `files` have any use in the context of a webpack loader? And if so, is there a way to get rid of it (or just ignore it/make it optional)?

**Comment by samvv** [comment]
Maybe this is a bit out of the scope of this project, but if what I'm saying makes sense, I'd be willing to give it a try.

**Comment by johnnyreilly** [comment]
Why don't you simply not supply `files` rather than supplying an empty one?  It's been a TypeScript compiler error to supply an empty files since TypeScript 2.1 (or thereabouts..)

**Comment by samvv** [comment]
Without a `files` object it defaults to fetching all the sources in the "current directory", which is the directory of the platform, not of the project I want my platform to compile 😕 That's why I said it might be out-of-scope, since it's not a very common use case.

**Comment by johnnyreilly** [comment]
Does the `configFileName` option not give you what you need? https://github.com/TypeStrong/ts-loader/tree/f4d8e711b2193cec358f94adf92202f39b8b2473#configfilename-string-defaulttsconfigjson

BTW doesn't your setup lead to a rather problematic experience in your editor of choice? It won't know what the settings in your `tsconfig.json` are and may report different errors than your build system.

**Comment by samvv** [comment]
@johnnyreily Not sure ... as I tried it with a patched version of ts-loader, it didn't work. And good point, I totally forgot about that! You're right, of course. Requiring a `tsconfig.json` solves it then. Thank you so much!
